cmd/access/node_builder/access_node_builder.go

```diff
@@ -680,39 +624,388 @@ func unstakedNetworkMsgValidators(log zerolog.Logger, idProvider id.IdentityProv
+				node.State,
+				engine.RequestCollections,
+				filter.HasRole(flow.RoleCollection),
+				func() flow.Entity { return &flow.Collection{} },
+			)
+			if err != nil {
+				return nil, fmt.Errorf("could not create requester engine: %w", err)
+			}
+
+			builder.IngestEng, err = ingestion.New(
+				node.Logger,
+				node.Network,
+				node.State,
+				node.Me,
+				builder.RequestEng,
+				node.Storage.Blocks,
+				node.Storage.Headers,
+				node.Storage.Collections,
+				node.Storage.Transactions,
+				node.Storage.Results,
+				node.Storage.Receipts,
+				builder.TransactionMetrics,
+				builder.CollectionsToMarkFinalized,
+				builder.CollectionsToMarkExecuted,
+				builder.BlocksToMarkExecuted,
+				builder.RpcEng,
+			)
+			if err != nil {
+				return nil, err
+			}
+			builder.RequestEng.WithHandle(builder.IngestEng.OnCollection)
+			builder.FinalizationDistributor.AddConsumer(builder.IngestEng)
+
+			return builder.IngestEng, nil
+		}).
+		Component("requester engine", func(node *cmd.NodeConfig) (module.ReadyDoneAware, error) {
+			// We initialize the requester engine inside the ingestion engine due to the mutual dependency. However, in
+			// order for it to properly start and shut down, we should still return it as its own engine here, so it can
+			// be handled by the scaffold.
+			return builder.RequestEng, nil
+		})
+
+	if builder.supportsObserverAndFollower {
+		builder.Component("unstaked sync request handler", func(node *cmd.NodeConfig) (module.ReadyDoneAware, error) {
+			syncRequestHandler, err := synceng.NewRequestHandlerEngine(
+				node.Logger.With().Bool("unstaked", true).Logger(),
+				unstaked.NewUnstakedEngineCollector(node.Metrics.Engine),
+				builder.AccessNodeConfig.PublicNetworkConfig.Network,
+				node.Me,
+				node.Storage.Blocks,
+				builder.SyncCore,
+				builder.FinalizedHeader,
+			)
+
+			if err != nil {
+				return nil, fmt.Errorf("could not create unstaked sync request handler: %w", err)
+			}
+
+			return syncRequestHandler, nil
+		})
+	}
 
-	if len(addresses) != len(keys) {
-		return nil, fmt.Errorf("number of addresses and keys provided for the boostrap nodes don't match")
+	if builder.executionDataSyncEnabled {
+		builder.BuildExecutionDataRequester()
 	}
 
-	ids := make([]*flow.Identity, len(addresses))
-	for i, address := range addresses {
-		key := keys[i]
+	builder.Component("ping engine", func(node *cmd.NodeConfig) (module.ReadyDoneAware, error) {
+		ping, err := pingeng.New(
+			node.Logger,
+			node.IdentityProvider,
+			node.IDTranslator,
+			node.Me,
+			builder.PingMetrics,
+			builder.pingEnabled,
+			builder.nodeInfoFile,
+			node.PingService,
+		)
+
+		if err != nil {
+			return nil, fmt.Errorf("could not create ping engine: %w", err)
+		}
+
+		return ping, nil
+	})
+
+	return builder.FlowNodeBuilder.Build()
+}
+
+// enqueueUnstakedNetworkInit enqueues the unstaked network component initialized for the staked node
+func (builder *FlowAccessNodeBuilder) enqueueUnstakedNetworkInit() {
```

Comment:
```suggestion
// enqueuePublicNetworkInit enqueues the public network component initialized for the staked node
func (builder *FlowAccessNodeBuilder) enqueuePublicNetworkInit() {
```

Reply:
Thank you for the review. The reason this was not addressed so far to make any merges from main easier until the code stabilizes.
Addressed: https://github.com/GetElastech/flow-go/commit/75a0d03f026929522b828e78208b4238b949e364